src/lib/modules/blockchain_connector/index.js

```diff
@@ -636,6 +650,16 @@ class BlockchainConnector {
     this.web3.eth.getTransaction(hash, cb);
   }
 
+  getRawTransactionHash(tx) {
+    return new Transaction(tx, 'hex').serialize().toString('hex');
+  }
+
+  getTransactionByRawTransactionHash(hash, cb) {
+    const rawData = Buffer.from(ethUtil.stripHexPrefix(hash), 'hex');
+    const tx = new Transaction(rawData, 'hex');
+    cb(null, tx.hash().toString('hex'));
```

Comment:
This doesn't seem to be the whole story yet. The produced hash is not the same as the one from the original transaction.